modules/shortcodes/flickr.php

```diff
@@ -174,15 +200,28 @@ function flickr_shortcode_video_markup( $atts ) {
 
 add_shortcode( 'flickr', 'flickr_shortcode_handler' );
 
-// Override core's Flickr support because Flickr oEmbed doesn't support web embeds
+// Override core's Flickr support because Flickr oEmbed doesn't support web embeds.
 wp_embed_register_handler( 'flickr', '#https?://(www\.)?flickr\.com/.*#i', 'jetpack_flickr_oembed_handler' );
 
+/**
+ * Callback to modify output of embedded Vimeo video using Jetpack's shortcode.
+ *
+ * @since 3.9
+ *
+ * @param array $matches Regex partial matches against the URL passed.
+ * @param array $attr    Attributes received in embed response.
+ * @param array $url     Requested URL to be embedded.
+ *
+ * @return string Return output of Vimeo shortcode with the proper markup.
+ */
 function jetpack_flickr_oembed_handler( $matches, $attr, $url ) {
-	// Legacy slideshow embeds end with /show/
-	// e.g. http://www.flickr.com/photos/yarnaholic/sets/72157615194738969/show/
+	/*
+	 * Legacy slideshow embeds end with /show/
+	 * e.g. http://www.flickr.com/photos/yarnaholic/sets/72157615194738969/show/
+	 */
 	if ( '/show/' !== substr( $url, -strlen( '/show/' ) ) ) {
-		// These lookups need cached, as they don't use WP_Embed (which caches)
-		$cache_key   = md5( $url . serialize( $attr ) );
+		// These lookups need cached, as they don't use WP_Embed (which caches).
+		$cache_key   = md5( $url . wp_json_encode( $attr ) );
```

Comment:
This does not output the exact same string as when using `serialize`, so images will need to be called and cached again. I could not think of another alternative to `serialize` that would output the exact same string.